File: Ejercicio 8/dispositivo.py
from abc import ABC,abstractmethod

class Encendible(ABC):
    @abstractmethod
    def encender(self): pass
    @abstractmethod
    def apagar(self): pass

class Dispositivo(ABC):
    def __init__(self,nombre):
        self.nombre=nombre

    @abstractmethod
    def info(self): pass

class TV(Dispositivo,Encendible):
    def __init__(self): super().__init__("TV")
    def info(self): print("Televisor:")
    def encender(self): print("TV encendida")
    def apagar(self): print("TV apagada")

class Radio(Dispositivo,Encendible):
    def __init__(self): super().__init__("Radio")
    def info(self): print("Radio:")
    def encender(self): print("Radio encendida")
    def apagar(self): print("Radio apagada")

class Celular(Dispositivo,Encendible):
    def __init__(self): super().__init__("Celular")
    def info(self): print("Celular:")
    def encender(self): print("Celular encendido")
    def apagar(self): print("Celular apagado")

dispositivos=[TV(),Radio(),Celular()]

for d in dispositivos:
    d.info()
    d.encender()
    d.apagar()

File: Ejercicio 8/dispositivo.java
interface Encendible{
    void encender();
    void apagar();
}

abstract class Dispositivo implements Encendible{
    String nombre;

    Dispositivo(String n){ nombre=n; }

    abstract void info();
}

class TV extends Dispositivo implements Encendible{
    TV(){ super("TV"); }

    void info(){ System.out.println("Televisor:"); }

    public void encender(){ System.out.println("TV encendida"); }
    public void apagar(){ System.out.println("TV apagada"); }
}

class Radio extends Dispositivo implements Encendible{
    Radio(){ super("Radio"); }

    void info(){ System.out.println("Radio:"); }

    public void encender(){ System.out.println("Radio encendida"); }
    public void apagar(){ System.out.println("Radio apagada"); }
}

class Celular extends Dispositivo implements Encendible{
    Celular(){ super("Celular"); }

    void info(){ System.out.println("Celular:"); }

    public void encender(){ System.out.println("Celular encendido"); }
    public void apagar(){ System.out.println("Celular apagado"); }
}

public class Main{
    public static void main(String[] args){

        Dispositivo[] dispositivos={
            new TV(),
            new Radio(),
            new Celular()
        };

        for(Dispositivo d:dispositivos){
            d.info();
            d.encender();
            d.apagar();
    }
}
}
